src/main/resources/db/changelog.xml

```diff
@@ -10,6 +10,7 @@
 
     <property name="identifier_type" value="varchar(63)"/>
     <property name="description_type" value="varchar(2047)"/>
+    <property name="dataset_snapshot_identifier_type" value="varchar(512)"/>
```

Comment:
I think to follow this pattern, we want `varchar(511)`, which makes sense to me (max should be 2^n=512, so we want 2^n-1). Thus, `DatasetSnapshotNameProperty` would have a `maxLength` of 511, and your test could try "a" repeated 512 times.

Reply:
Updated!

Reply:
Also is there a particular reasoning behind using this pattern?